include/zephyr/net/net_context.h

```diff
@@ -348,8 +348,14 @@ __net_socket struct net_context {
 
 	/** IPv6 hop limit or IPv4 ttl for packets sent via this context. */
 	union {
-		uint8_t ipv6_hop_limit;
-		uint8_t ipv4_ttl;
+		struct {
+			uint8_t ipv6_hop_limit;
+			uint8_t ipv6_mcast_hop_limit;
+		};
+		struct {
+			uint8_t ipv4_ttl;
+			uint8_t ipv4_mcast_ttl;
+		};
```

Comment:
would be nice to have doxygen comments here as well (again, unless this is something you want to hide from the public API).

Reply:
Sure, I will submit a follow up pr adding more comments. Thx!